Fix early application of correlated subqueries (#13593)
crate/crate: docs/appendices/release-notes/unreleased.rst

.. stable branch. You can add a version label (`v/X.Y`) to the pull request for
.. an automated mergify backport.

> - Fixed an issue that prevented correlated sub-queries from accessing the
>   multiple joined outer query and caused ``IllegalStateException``.
>   An example ::

Reviewer: ```suggestion
- Fixed an issue that caused correlated sub-queries to fail with an `IllegalStateException` when the outer query contained multiple joins. An example ::
```

Author: Thank you!
